lib/src/serializable.dart

```diff
@@ -0,0 +1,237 @@
+  void handleEvent(CustomEvent apiCall) {
+    _eventController.add(apiCall.detail);
+  }
+
+  JsObject get wkBridge {
+    if (context['webkit'] != null) {
+      return context['webkit']['messageHandlers']['bridge'];
+    } else {
+      return null;
+    }
+  }
+}
+
+class SerializableEvent<T> extends Event<T> {
+  SerializableEvent(this._eventKey, DispatchKey dispatchKey)
+      : super(dispatchKey);
+
+  String _eventKey;
+  String get eventKey => _eventKey;
+}
+
+abstract class SerializableEvents {
+  List<SerializableEvent> get allEvents => [];
+}
+
+abstract class SerializableModule extends Module {
+  SerializableModule() {
+    _registerWithSerializableBus();
+  }
+
+  void _registerWithSerializableBus() {
+    SerializableBus.sharedBus.registerModule(this);
+  }
+
+  @override
+  SerializableEvents get events => null;
+
+  String get serializableKey => null;
+}
+
+class SerializableBus {
+  SerializableBus() {
+    _registeredModules = new Map<String, SerializableModule>();
+    _logger = new Logger('Serializable Bus');
+  }
+
+  static final SerializableBus _singleton = new SerializableBus();
+  static SerializableBus get sharedBus => _singleton;
+
+  Bridge _bridge;
+  Logger _logger;
+  Map<String, SerializableModule> _registeredModules;
+
+  void reset() {
+    _registeredModules = new Map<String, SerializableModule>();
+    _bridge = null;
+  }
+
+  void registerModule(SerializableModule module) {
+    if (module.serializableKey != null) {
+      _registerForLifecylceEvents(module);
+      _registeredModules[module.serializableKey] = module;
+    } else {
+      _logger.warning('Unable to serialize module without serializableKey');
+    }
+  }
+
+  void deregisterModule(SerializableModule module) {
+    _registeredModules.remove(module.serializableKey);
+  }
+
+  void _registerForAllEvents(SerializableModule module) {
+    if (module.events != null) {
+      for (var event in module.events.allEvents) {
+        if (event is SerializableEvent) {
+          event
+              .listen((payload) => _sendEvent(module, event.eventKey, payload));
+        }
+      }
+    } else {
+      _logger
+          .warning('Events not defined for ${module.serializableKey} module');
+    }
+  }
+
+  void _registerForLifecylceEvents(SerializableModule module) {
+    module.willLoad.listen((_) {
+      _registerForAllEvents(module);
+      _sendEvent(module, 'willLoad', null);
+    });
+    module.didLoad.listen((_) => _sendEvent(module, 'didLoad', null));
+    module.willUnload.listen((_) => _sendEvent(module, 'willUnload', null));
+    module.didUnload.listen((_) {
+      _sendEvent(module, 'didUnload', null);
+      deregisterModule(module);
+    });
+  }
+
+  void _sendEvent(SerializableModule module, String eventKey, Object data) {
+    Map dataToSend = new Map();
+    dataToSend['module'] = module.serializableKey;
+    dataToSend['event'] = eventKey;
+
+    if (data is JsonSerializable) {
+      data = (data as JsonSerializable).toJson();
```

Comment:
Okay, it's fine as is then. Out of curiosity - what version of the Dart SDK are you on?